### ISSUE TITLE
Add support for Laravel .blade.php templates

### ISSUE THREAD

**Issue by calvinchoy** [Glavin001/atom-beautify#448]
Hi,

Anyone know where I can add the .blade.php of laravel blade templating so that beautifier recognise it? For example format it like html, which might work out of the box. 

Thanks

**Comment by Glavin001** [comment]
- Please provide an example to be added to the tests
- Does PHP work? The grammar and extension could be added here: https://github.com/Glavin001/atom-beautify/blob/master/src/languages/php.coffee#L9-L18
- If it's not pure HTML or PHP then I would want to add it as a new language. See https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md#how-to-add-a-language for details.

I am always open to reviewing Pull Requests.

**Comment by calvinchoy** [comment]
I added the `Blade` grammar and `.blade.php` to the `html.coffee` file. The formatting worked actually pretty good except for some cases. 

I will try to fix those problems.

**Comment by Glavin001** [comment]
`html` is handled by `JS Beautify` beautifier by default. This means we should be able to add `Blade` language and then use `JS Beautify` and quickly get going.

I will review and accept Pull Requests for this. See https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md#how-to-add-a-language for documentation. Feel free to ask any questions. Thanks.

**Comment by bojidar-bg** [comment]
:+1:

**Comment by kmare** [comment]
any news on that? it'd be quite cool..

**Comment by ubeyou** [comment]
Thanks calvinchoy, it works! 

uses Auto-Intent before Beautify, it works great on my side.

add the `Blade` grammar and `php` to the `html.coffee` file. 

```
  ###
  Supported Grammars
  ###
  grammars: [
    "HTML",
    "Blade"
  ]

  ###
  Supported extensions
  ###
  extensions: [
    "html",
    "php"
  ]
```

**Comment by ArtworkAD** [comment]
👍 where is the html.coffee? would like to see this merged

**Comment by Glavin001** [comment]
@ArtworkAD : This is the `html.coffee` file: https://github.com/Glavin001/atom-beautify/blob/master/src/languages/html.coffee#L17-L26

**Comment by goemaeret** [comment]
Any update on this one?

**Comment by slottybon** [comment]
Yeah, any update? 
Using the `html.coffee` file does work for the html part in my `blade.php` files, but obviously not for the blade part.

Meaning this:
```
@section('scripts')
	{{ HTML::script('http://code.highcharts.com/highcharts.js') }}
	{{ HTML::script('js/app/nav.js') }}
	{{ HTML::script('js/app/stream/scrollTop.js') }}
	{{ HTML::script('js/jquery-sticky/jquery.sticky.js') }}
	{{ HTML::script('js/app/partials/sidebar.js') }}
@stop
```
becomes this: 
```
@section('scripts') {{ HTML::script('http://code.highcharts.com/highcharts.js') }} {{ HTML::script('js/app/nav.js') }} {{ HTML::script('js/app/stream/scrollTop.js') }} {{ HTML::script('js/jquery-sticky/jquery.sticky.js') }} {{ HTML::script('js/app/partials/sidebar.js') }} @stop
```
So, yeah ... did anybody get further than this?

**Comment by whereisjefe** [comment]
Just wanted to add to what @slottybon said, if there is a temp fix that would be great

**Comment by Sangramchahal** [comment]
when can we expect to see built in support for blade.php files?

**Comment by maxvaser** [comment]
I love this library! Format on save is fantastic! Adding formatting support for the Blade syntax and HTML would be very useful!  or even just recognizing  .blade.php  file extensions to allow the regular HTML portion to be formatted on save would be awesome!

 I tried modifying the html.coffee file and it had no effect.

**Comment by Glavin001** [comment]
> when can we expect to see built in support for blade.php files?

As soon as someone submits a Pull Request implement it 😜.

See https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md for how to add a language and a beautifier to Atom-Beautify.
Let me know if you have any questions! It looks like @ubeyou commented what was necessary: https://github.com/Glavin001/atom-beautify/issues/448#issuecomment-191680738
However, I would prefer to add a new language (separate from existing HTML) for Blame as it is not simply HTML.

For example:

```coffeescript
module.exports = {

  name: "Blade"
  namespace: "blade"
  fallback: ["html"]

  ###
  Supported Grammars
  ###
  grammars: [
    "HTML",
    "Blade"
  ]

  ###
  Supported extensions
  ###
  extensions: [
    "html",
    "php"
  ]

}
```

**Comment by nckrtl** [comment]
Would really love blade support as well!

**Comment by ryaposov** [comment]
Blade support would be a great thing!

**Comment by sogeniusio** [comment]
Waiting for some support as well...

**Comment by Glavin001** [comment]
@nckrtl @ryaposov @sogeniusio : The power is yours 😜 . Pull Requests are welcome! See https://github.com/Glavin001/atom-beautify/issues/448#issuecomment-270036847 for details. If you have questions about how to make the Pull Request, please let me know.

Also, to new users, please do not create unproductive comments for "+1" or similar, and instead *do* put your 👍  on the Issue body at the top: 
![image](https://cloud.githubusercontent.com/assets/1885333/25815812/74009ba8-33f8-11e7-89ae-b6dddd1f8ee0.png)
Thank you!

**Comment by 1e4** [comment]
Has there been any more progress by anyone on this? Judging by the thread, the temp fix seems to have some pitfalls.

**Comment by Glavin001** [comment]
@1e4 no progress planned on my end. I await a Pull Request. Thanks!

**Comment by jhkueh** [comment]
Hi, I had submitted a Pull Request (#1817) to add basic support for this not too long ago.

The problem is that Blade syntax don't indent properly yet. I am hoping if anyone could point me to the right direction or give me some tips, so I could further work on it.

**Comment by szeck87** [comment]
Disregard the previous comment.

**Comment by hosquiat** [comment]
Whew, thanks!

**Comment by DevBodin** [comment]
I'd be interested in helping out with this if anyone's got something going. Not very experienced with Coffeescript (or Github for that matter), but with a little hand-holding I could make it happen.

**Comment by jhkueh** [comment]
@DevBodin I had submitted a PR (#1817) for this that add basic support a few months ago. It still need some works before it's accepted, which I had been unable to do due to the busyness of life. Since it's holiday season now, I will be able to resume working on it.

**Comment by darron1217** [comment]
@jhkueh Will you send a PR for this issue again?

**Comment by jhkueh** [comment]
@darron1217 feel free to pick up from where I left off in this PR (#1817), if possible. Because I no longer use Atom as it doesn't suit my needs, I have since moved on to VS Code now.

**Comment by darron1217** [comment]
@jhkueh Does VS Code support blade template well ?

**Comment by ahmedalmulki** [comment]
Hi is there any news about supporting blade , I am really need that .

**Comment by darron1217** [comment]
@ahmedalmulki I've created PR #2087 and passed tests, but not yet merged...
Due to lack of knowledge about beautifying, I could just implement basic level (just arranging HTML) of Blade Template.
If anyone can make my PR better, please let me know :)

**Comment by HDVinnie** [comment]
please add blade support

**Comment by HexagonStorms** [comment]
Pretty frustrating seeing that this issue is 2 & a half years old and still lacking Blade support... Especially since there is an open pull request that we're just waiting on to be merged into master.

**Comment by darron1217** [comment]
@Glavin001 Ping

**Comment by HDVinnie** [comment]
@HexagonStorms I agree and its not like blade is not widely used by many people. Laravel is quiet popular.

**Comment by szeck87** [comment]
Published to v0.32.3.

**Comment by shawnfromportland** [comment]
> @HexagonStorms I agree and its not like blade is not widely used by many people. Laravel is quiet popular.

@HexagonStorms 😂😂😂
